packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -75,6 +75,12 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, Semver {
     /// @notice An internal mapping to allow for constant-time lookups of existing claims.
     mapping(ClaimHash => bool) internal claims;
 
+    /// @notice An internal mapping of subgames rooted at a claim index to other claim indices in the subgame.
+    mapping(uint256 => uint256[]) internal subgames;
```

Comment:
Subgames have a max depth of 1. So this maps claim indices to an array of other claims that _directly_ dispute it.
The total number of subgames and their edges are both $O(N)$.